docker/ubuntu_22.04-dev.dockerfile

```diff
@@ -65,30 +74,52 @@ RUN conda uninstall -y conda-libmamba-solver
 RUN conda config --set solver classic
 RUN conda update -y --all && \
     mamba install -y \
-                cmake \
+                make \
+                gxx_linux-64 \
+                gcc_linux-64 \
                 git \
+                cmake \
+                gfortran \ 
                 libblas \
                 liblapack \
+                eigen \
                 hdf5 \
-                setuptools \
+                numpy==1.23 \
+                scipy \
+                "cython<3" \
+                nose \
                 pytest \
                 pytables \
+                matplotlib \
                 jinja2 \
-                "cython<3" \
+                setuptools \
+                future \
+                progress \
                 && \
     mamba install -y --force-reinstall libsqlite && \
     conda clean -y --all
 RUN mkdir -p `python -m site --user-site`
 
+ENV CC /opt/conda/bin/x86_64-conda_cos6-linux-gnu-gcc
+ENV CXX /opt/conda/bin/x86_64-conda_cos6-linux-gnu-g++
+ENV CPP /opt/conda/bin/x86_64-conda_cos6-linux-gnu-cpp
+
+# put conda on the path
+ENV LD_LIBRARY_PATH /opt/conda/lib:$LD_LIBRARY_PATH
+# set HDF5 path for base_python stage
+ENV HDF5_INSTALL_PATH=/opt/conda
+
+
 FROM ${pkg_mgr}_deps AS base_python
 # make starting directory
 RUN mkdir -p $HOME/opt
 RUN echo "export PATH=$HOME/.local/bin:\$PATH" >> ~/.bashrc
 
 # build HDF5
 ARG build_hdf5="NO"
-ENV HDF5_INSTALL_PATH=$HOME/opt/hdf5/$build_hdf5
-RUN if [ "$build_hdf5" != "NO" ]; then \
+ARG pkg_mgr=apt
+# don't build hdf5 if we use conda (we already installed it)
+RUN if [ "$build_hdf5" != "NO" ] && [ "$pkg_mgr" = "apt" ]; then \
```

Comment:
I think we may move this to the apt_deps stage. Thus, there is no need to use extra conditions.

Reply:
We could do that, but then there would be almost nothing left in the `base_python` stage, which means that we should probably remove it. However, that would require some more edits in the `docker_publish.yml` file to accomodate the new stages. Overall, I think there needs to be more discussion on this topic, and maybe we can also consider splitting the dockerfile into two dockerfiles (one for conda, one for apt).

Reply:
@gonuke, we are expecting your valuable suggestions on this.

Reply:
My opinion, employing a single Dockerfile with a multi-stage build is a viable approach. It may necessitate a redesign of the `docker_publish.yml`. Additionally, I've observed that we are using the ubuntu base image for each build, resulting in larger image sizes. To optimize this, we can create `apt_deps` from the debian image, as we do not need snap support. For `conda_deps`, we can initiate the build from scratch or use [miniconda3](https://hub.docker.com/r/continuumio/miniconda3).

Reply:
@gonuke, what is your opinion on this?

Reply:
Let's stick with the Ubuntu as the base image since it also confirms that we can build on that image.